src/analyzing/AST_to_IL.ml

```diff
@@ -1545,10 +1537,11 @@ and stmt_aux env st =
       ss @ stmt env stmt1
   (* Rust: unsafe block *)
   | G.OtherStmtWithStmt (G.OSWS_Block ("Unsafe", tok), [], stmt1) ->
-      let todo_stmt =
-        mk_s (Instr (mk_fake_TODO_instr ~str:"unsafe_block" env tok))
-      in
-      todo_stmt :: stmt env stmt1
+      let todo_stmt = fixme_stmt ToDo (G.TodoK ("unsafe_block", tok)) in
+      todo_stmt @ stmt env stmt1
+  | G.OtherStmt (OS_Async, [ G.S stmt1 ]) ->
```

Comment:
we might want to promote OS_Async to a proper stmt if it's a statement in most languages,
or to move it in OtherStmtWithStmt?

Reply:
According to this comment

```ocaml
  (* any here should _not_ contain any statement! otherwise the CFG will be
   * incorrect and some analysis (e.g., liveness) will be incorrect.
```

I think yes we should move it in `OtherStmtWithStmt`.

Reply:
I left a TODO note for now, I just prefer to have that change in a different PR. If you think that is urgent I can try open one this week.